add: readme check/remindにfilter機能を追加しました。

ganondorofu opened the pull request:
5個までロールによるフィルターができるようにしました。
andとor条件で検索できます。

saitogo555 committed:
Initial commit

saitogo555 committed:
feat(tsconfig): TypeScript設定ファイルを追加

saitogo555 committed:
feat(package): package.jsonとpackage-lock.jsonを追加

saitogo555 committed:
feat(package): package.jsonとpackage-lock.jsonを更新

- package.jsonにbiome関連のスクリプトを追加
- package.jsonのdevDependenciesに@biomejs/biomeを追加
- package-lock.jsonを更新し、@biomejs/biomeの依存関係を追加

saitogo555 committed:
feat(biome): biome.jsonを追加

saitogo555 committed:
feat(scripts): startスクリプトを追加

saitogo555 committed:
feat(bot): Discordボットの初期設定を追加

saitogo555 committed:
feat(config): 環境変数の設定を追加

- dotenvを使用して環境変数をロードする機能を追加
- DISCORD_TOKENとCLIENT_IDをconfig.tsからインポート
- index.tsからdotenvの設定を削除し、config.tsを利用するように変更

saitogo555 committed:
fix(config): 環境変数のデフォルト値を設定

- DISCORD_TOKENとCLIENT_IDにデフォルト値を追加

saitogo555 committed:
feat(event): メッセージ作成イベントハンドラーを追加

- 新しいメッセージ作成イベントハンドラーを実装
- メッセージがボットからのものでないか確認
- ギルドが存在するか確認
- 特定のクライアントIDがメンションされているか確認
- メンションされたユーザーを取得し、返信メッセージを送信
- index.tsにメッセージ作成イベントハンドラーを登録

saitogo555 committed:
feat(scripts): watchスクリプトを追加

- src/index.tsを監視するためのwatchスクリプトを追加しました。

saitogo555 committed:
refactor(event): ユーザー抽出の処理条件を修正

saitogo555 committed:
refactor(event): メッセージ作成イベントハンドラーのコードを簡素化

- 不要なコメントを削除
- 条件文を一行に統合して可読性を向上

saitogo555 committed:
refactor(event): メッセージ作成イベントハンドラーのコメントを日本語に翻訳

saitogo555 committed:
fix(config): 環境変数のデフォルト値を設定

- DISCORD_TOKENとCLIENT_IDにデフォルト値を追加

saitogo555 committed:
Merge branch 'feature/message-detection'

saitogo555 committed:
refactor(index): エラーメッセージとログメッセージを日本語に変更

saitogo555 committed:
refactor(config): 環境変数の読み込み処理を改善

- コメントを日本語に翻訳
- 定数のエクスポートを一貫性のある形式に修正

saitogo555 committed:
feat(event): メッセージ作成イベントハンドラーにリアクション機能を追加

- READ_REACTION_EMOJIをインポート
- メッセージに対してリアクションを追加する処理を実装
- 不要なconsole.logとreply処理を削除

saitogo555 committed:
refactor(event): getTargetUsers関数をutilsに移動

- メッセージ作成イベントハンドラーからgetTargetUsers関数を分離
- コードの可読性と再利用性を向上

saitogo555 committed:
feat(event): メッセージ更新イベントハンドラーを追加

- Discordのメッセージ更新イベントを処理する関数を実装
- メッセージが更新された際にリアクションを追加する機能を追加

saitogo555 committed:
refactor(event): メッセージ更新イベントハンドラーからリアクション処理を削除

- メッセージ更新時のリアクション処理を削除しました。
- コードの可読性を向上させるため、不要な行を整理しました。

saitogo555 committed:
feat(client): GuildMembersおよびGuildPresencesのインテントを追加

- DiscordクライアントのインテントにGuildMembersとGuildPresencesを追加しました
- これにより、メンバーの情報やプレゼンスの状態を取得できるようになります

saitogo555 committed:
refactor(utils): getTargetUsers関数を非同期に変更し、メンション処理を改善

- getTargetUsers関数をasyncに変更し、Promiseを返すように修正
- ロールメンションから全ての対象ユーザーを取得する処理を明確化
- @everyoneまたは@hereメンション時の処理を追加し、オンラインメンバーのフィルタリングを実装

saitogo555 committed:
refactor(event): getTargetUsers関数を非同期に変更

- messageCreateEventHandlerおよびmessageUpdateEventHandler内でgetTargetUsersを非同期に呼び出すように変更

saitogo555 committed:
style(event, utils): コメントのフォーマットを修正

saitogo555 committed:
refactor(utils): getTargetUsers関数からデバッグ用のconsole.logを削除

- 不要なログ出力を削除し、コードをクリーンにしました

saitogo555 committed:
Merge branch 'feature/message-detection'

saitogo555 committed:
feat(message): isTargetMessage関数を追加し、メッセージの対象判定を実装

- メッセージがボットによって送信されていないか確認
- ギルド内で送信されているか確認
- ボット自身がメンションされているか確認
- メンションされた対象ユーザーが存在するか確認

saitogo555 committed:
refactor(utils): OmitPartialGroupDMChannel型を削除し、Message型に統一

- isTargetMessage関数とgetTargetUsers関数の引数型を変更
- コードの可読性と一貫性を向上

saitogo555 committed:
feat(event): メッセージリアクション追加イベントハンドラーを実装

- Discordのメッセージリアクション追加イベントを処理する関数を追加
- リアクションが追加されたメッセージを取得し、対象メッセージかどうかを判定
- 特定のリアクション以外が追加された場合にユーザーのリアクションを削除

saitogo555 committed:
Merge branch 'feature/reaction-detection'

saitogo555 committed:
refactor(event): メッセージ作成および更新イベントハンドラーのリファクタリング

- 不要な条件チェックを削除し、isTargetMessage関数を使用して対象メッセージを判定
- コードの可読性を向上させるために、重複したロジックを統一

saitogo555 committed:
refactor: コードの整形

saitogo555 committed:
feat(config): COMMAND_NAME環境変数の追加

- .envファイルからCOMMAND_NAMEを読み込むように変更
- デフォルト値は「readme」

saitogo555 committed:
feat(event): interactionCreateイベントハンドラーを追加

- interactionCreateイベントを処理する新しいハンドラーを実装
- 指定されたコマンド名が実行された際に応答を返す機能を追加

saitogo555 committed:
feat(command): helpコマンドを追加

- ヘルプコマンドを実装し、利用可能なコマンドの一覧を表示
- interactionCreateイベントハンドラーにサブコマンドの処理を追加

saitogo555 committed:
feat(config): カラー定数を追加

- INFO_COLOR, SUCCESS_COLOR, WARNING_COLOR, ERROR_COLORを定義

saitogo555 committed:
feat(help): INFO_COLORを使用してヘルプコマンドの色を設定

- INFO_COLORをconfigからインポート
- ヘルプコマンドの色設定を定数に変更

saitogo555 committed:
feat(command): スラッシュコマンドの登録機能を追加

- discord.jsからRESTとRoutesをインポート
- CLIENT_IDを設定ファイルからインポート
- BOTの起動時にスラッシュコマンドを登録する処理を追加

saitogo555 committed:
feat(command): checkコマンドハンドラーを追加し、リマインダーの説明を更新

- checkCommandHandlerをcommandsに追加
- remindサブコマンドの説明を「指定メッセージの未読者にDMでリマインダーを送信」に変更

saitogo555 committed:
feat(command): checkコマンドハンドラーを追加

- 指定メッセージの既読状況を確認する機能を実装
- メッセージIDが指定されていない場合のエラーハンドリングを追加
- チャンネルが見つからない場合のエラーハンドリングを追加
- 指定されたメッセージが見つからない場合のエラーハンドリングを追加
- リアクションしたユーザーを取得し、既読・未読ユーザーを分類
- 結果をEmbed形式で表示

saitogo555 committed:
Merge branch 'feature/command'

saitogo555 committed:
feat(embed): エラーメッセージ用のEmbedを構築する関数を追加

saitogo555 committed:
refactor(command): エラーメッセージの構築をbuildErrorEmbed関数に統一

- エラーメッセージの構築を一貫性のあるbuildErrorEmbed関数を使用するようにリファクタリング
- 不要なERROR_COLORのインポートを削除

saitogo555 committed:
feat(embed): 成功および情報用のEmbedを構築する関数を追加

- buildSuccessEmbed関数を追加し、成功メッセージを構築
- buildInfoEmbed関数を追加し、情報メッセージを構築

saitogo555 committed:
feat(command): remindコマンドハンドラーを追加

- 指定メッセージの未読者にDMでリマインダーを送信する機能を実装
- メッセージIDを取得し、未読者にリマインドメッセージを送信
- エラーハンドリングを追加し、適切なエラーメッセージを返信

saitogo555 committed:
Merge branch 'feature/command'

saitogo555 committed:
feat(interaction): コマンドが存在しない場合のエラーメッセージをEmbedで通知

- buildInfoEmbedを使用して、ユーザーにコマンドの存在しないことを通知

saitogo555 committed:
refactor: インポート順序を整理し、コードの可読性を向上

saitogo555 committed:
feat(activity): BOTのアクティビティを設定する機能を追加

- Discord BOTが起動時にスラッシュコマンドのヘルプを表示するアクティビティを設定

saitogo555 committed:
refactor(command): エイリアスを削除

- check, help, remindコマンドからエイリアスを削除しました

saitogo555 committed:
refactor(event): 権限チェックの追加

- ユーザーの権限を確認するロジックを追加

saitogo555 committed:
feat(command): usageコマンドハンドラーを追加

- ReadmeBotの使い方を表示するコマンドを実装
- コマンド一覧にusageコマンドを追加
- ユーザーガイドをEmbed形式で提供

saitogo555 committed:
Merge branch 'feature/command'

saitogo555 committed:
refactor(usage): 不要な改行を削除し、フッターを削除

- 説明文の改行を削除し、読みやすさを向上
- フッターのメッセージを削除

saitogo555 committed:
refactor(command): エイリアスのチェックを削除

- コマンドの実行時にエイリアスを考慮しないように変更

saitogo555 committed:
feat(readme): README.mdを追加

saitogo555 committed:
refactor: コメントを整理

- 不要なコメントを削除
- コメント内容を簡潔に修正

saitogo555 committed:
docs(readme): README.mdのタイトルとリンクを修正

- タイトルから絵文字を削除
- リンクの絵文字を削除

saitogo555 committed:
docs(usage): 基本的な流れの説明を削除

saitogo555 committed:
feat(dependencies): @hono/node-serverとhonoを追加

- @hono/node-serverのバージョン1.14.3を追加
- honoのバージョン4.7.11を追加

saitogo555 committed:
feat(server): ヘルスチェックサーバーを追加

- 新しいサーバー設定を追加し、ヘルスチェック用のエンドポイントを実装
- 環境変数からポート番号を取得し、サーバーを起動

saitogo555 committed:
refactor(server): ヘルスチェックエンドポイントのレスポンスを整理

- "name"フィールドを"message"に変更
- "platform"と"memory_usage"フィールドを削除

saitogo555 committed:
Merge branch 'feature/health-check-for-koyeb'

saitogo555 committed:
feat(dependencies): node-cronを追加

saitogo555 committed:
feat(cron): ヘルスチェックの定期実行機能を追加

- node-cronを使用して10分ごとにヘルスチェックを実行
- ヘルスチェックの結果をコンソールにログ出力
- エラー発生時のハンドリングを追加

saitogo555 committed:
Merge branch 'feature/health-check-for-koyeb'

saitogo555 committed:
feat(cron): ヘルスチェックのログにURLを追加

- ヘルスチェック実行中のログにHEALTH_CHECK_URLを表示
- cronスケジュールの表現を修正

saitogo555 committed:
fix(checkCommand): 返信方法を修正

- 一貫性のあるエラーレスポンスのために、replyメソッドをeditReplyに変更

saitogo555 committed:
fix(client): パーシャルのインポート

saitogo555 committed:
feat(messageUtils): ギルド内のメッセージ検索機能を追加

- ギルド内のすべてのチャンネルから指定されたメッセージIDを持つメッセージを検索する関数を追加
- 既存のメッセージ取得機能を補完

saitogo555 committed:
feat(checkCommand): 既読状況確認コマンドにページネーション機能を追加

- メッセージ取得方法をチャンネルからギルドに変更
- 既読・未読ユーザーのページネーションを実装
- 結果表示用のEmbedを改善
- インタラクションハンドラーを追加し、ボタン操作に対応

saitogo555 committed:
feat(checkCommand): エラーハンドリングを強化し、応答メッセージを改善

- 既読状況確認コマンドの応答処理をリファクタリング
- メッセージID未指定時のエラーメッセージを改善
- ページネーションのインタラクションハンドラーにエラーハンドリングを追加

saitogo555 committed:
feat(remindCommand): リマインダーコマンドのエラーハンドリングを強化

- インタラクションの初期応答にエラーハンドリングを追加
- メッセージ取得処理を改善し、エラーメッセージを明確化
- 未読者へのDMリマインダー送信処理を整理

saitogo555 committed:
feat(remindCommand): 管理者権限のチェックを追加

- コマンド実行時にユーザーの管理者権限を確認
- 権限がない場合はエラーメッセージを送信

saitogo555 committed:
refactor(interactionCreateEvent): 実行条件を簡素化

- 不要なPermissionFlagsBitsのインポートを削除
- 管理者権限のチェックを削除し、ギルドの存在のみを確認

saitogo555 committed:
fix(remindCommand): リマインダー送信時のエラーメッセージを改善

- ユーザーの表示名を用いてエラーメッセージをログに記録するように変更

saitogo555 committed:
refactor(remindCommand, checkCommand): コメントの整理

- 管理者権限チェック、メッセージID取得、サーバー取得、メッセージ取得のコメントを追加
- コメントの表現を統一

saitogo555 committed:
Merge branch 'fix/unknow-interaction-error'

saitogo555 committed:
fix(remindCommand): 管理者権限が必要であることを明示するエラーメッセージに修正

saitogo555 committed:
refactor: コード整形

saitogo555 committed:
refactor(remindCommand): 管理者権限チェックのロジックを簡素化

- 管理者権限の確認を変数に格納し、可読性を向上

saitogo555 committed:
feat(remindCommand): 投稿者に実行権限付与

- 権限チェックをメッセージ取得後に移動
- コマンド実行の条件を管理者またはメッセージの投稿者に変更

saitogo555 committed:
refactor(remindCommand): 権限チェックのコメントを修正

saitogo555 committed:
Merge pull request #1 from saitogo555/feature/remind-execution-permission

feat(remind): remindコマンドを管理者と投稿者が実行できるように変更

saitogo555 committed:
refactor(events): イベントハンドラーのコメントを修正

- メッセージ作成、更新、リアクション追加の処理に関するコメントを明確化
- 不要な説明を削除し、関数の目的を簡潔に記述

saitogo555 committed:
docs(README): 環境変数の説明を明確化し、コマンドの権限を更新

- `.env` ファイルの説明を修正
- コマンドの権限を「全員」と「管理者, 投稿者」に変更

saitogo555 committed:
Merge pull request #2 from saitogo555/chore/update-comments

chore: コメントとドキュメントを修正

saitogo555 committed:
docs(README): セクションタイトルの絵文字を削除し、整形を改善

- セクションタイトルから絵文字を削除
- 一貫性のあるフォーマットに修正

saitogo555 committed:
docs(README): セクションリンクのフォーマットを修正

saitogo555 committed:
Merge pull request #3 from saitogo555/docs/update-style

ganondorofu committed:
chore: tsxのバージョンを4.20.3に更新

ganondorofu committed:
feat: 既読状況確認コマンドにロールフィルターオプションを追加

ganondorofu committed:
feat: リマインダーコマンドにロールフィルターオプションを追加

- remindCommand.ts: リマインダーコマンドにロールフィルターを適用するロジックを追加し、未読ユーザーの取得時にフィルターを考慮するように変更。
- index.ts: スラッシュコマンドデータにロールフィルターオプションを追加。
- config.ts: GUILD_IDをインポートし、開発用ギルドIDの設定を追加。

ganondorofu committed:
feat: 複数のロールフィルターオプションとフィルター条件を追加

checkCommand.tsとremindCommand.tsにおいて、複数のロールフィルターを取得できるように変更しました。フィルター条件としてAND/ORを選択できるオプションを追加し、フィルター適用時の動作を更新しました。これにより、ユーザーは指定したロールに基づいてフィルタリングを行うことが可能になります。

ganondorofu committed:
fix: コードの整形とフィルター条件のロジックを改善

- 不要な空行を削除
- フィルター条件のOR条件を明確にし、コードの可読性を向上

saitogo555 commented:
フィルターオプションを`filter1`~`filter5`まで作っていますが、これを`filter_mode`というオプションにして、`OR`か`AND`を選べるようにしてください。`filter_mode`が指定されたときはコマンド実行後にロール選択ボックスが自分にのみ表示され、ロールを選択してOKボタンを押すと、選択したロールだけにコマンドの処理が適用されるようにしてください。

ganondorofu committed:
feat: ロールフィルターオプションを追加し、インタラクション処理を改善

- biome.json: suspiciousルールを追加
- checkCommand.ts: ロールフィルターの選択メニューを追加し、フィルター条件の取得方法を改善
- remindCommand.ts: ロールフィルターの選択メニューを追加し、リマインダー送信処理を改善
- index.ts: コードのコメントを追加し、ヘルスチェックサーバーの起動に関するログを改善
- interactionCreateEvent.ts: インタラクション処理のエラーハンドリングを改善し、ボタンおよび選択メニューの処理を追加
